### PR TITLE
Add Discrete Fourier Transform functions to Numpy frontend

### PR DESCRIPTION
**Pull request by MartinOcharo**
ifft2 #19055

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by V-G-spec** [comment]
Hey, I see branch conflicts with your PR. Can you resolve those please?

**Comment by V-G-spec** [comment]
Hi @MartinOcharo thanks for resolving the conflicts. However, 13 tests are still failing. Can you take a look at those?

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by MartinOcharo** [comment]
Hi @V-G-spec, let me rectify the 13 tests that are still failing.

**Comment by AlexandraPosta** [comment]
Hello! Great work so far, just a few issues to sort:

- Would you be able to revert any changes made to other functions apart from ifft2? (the decorator was removed from fftfreq, some changes to the test routine for fftfreq)
- You have added the test for ifft2 twice and, for the second instance, misspelled the name of the function to "test_numpy_iftt2". It would be ideal to keep a single implementation with the correct name
- Have a look over the log for the failed tests

**Comment by MartinOcharo** [comment]
Thanks, @AlexandraPosta. Let me work on the issues.

**Review by MartinOcharo — COMMENTED** [review]
Re-added  the decorator back for fftfreq

**Comment by AlexandraPosta** [comment]
Hello @MartinOcharo, it seems like you are missing an argument for the test ('backend_to_test'). Can you have a look at the raw logs where the tests are failing and see if anything else is missing? Thank you!

**Comment by MartinOcharo** [comment]
@AlexandraPosta I remove my script when I tried to commit. The number of failed tests is still the same -11. What might be the issue since with or without my added code there are fails?

**Comment by AlexandraPosta** [comment]
Hello @MartinOcharo, I have checked the logs as well and none of them are related to your function. We get these errors sometimes because multiple functions are checked in the CI pipeline (for more information go [here](https://unify.ai/docs/ivy/overview/deep_dive/continuous_integration.html)), not only your function. So as long as none of the fails are related to your function, then we can merge your implementation. 

Now, let's have another look at your implementation, maybe push it back to check that all the test function's parameters are placed correctly. Thank you for your effort so far! :)

**Comment by MartinOcharo** [comment]
Hello @AlexandraPosta. Kindly check the fails from your end.

**Comment by AlexandraPosta** [comment]
Hello @MartinOcharo! You have not defined 'backend_fw', please have a look at the error messages. Go to the tests and press on "Details" for the tests that fail. Then search for "FAILURE" on the raw logs. You should find failures that are related to your function and change your implementation accordingly. Hope this helps! :)
image

**Comment by AlexandraPosta** [comment]
Hello! Please solve the merge conflicts and I will check the code right after :)

**Comment by MartinOcharo** [comment]
Hello @AlexandraPosta! Please have a look and revert.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_fft/test_discrete_fourier_transform.py [hunk 1]
```diff
@@ -254,3 +255,33 @@ def test_numpy_rfftn(dtype_and_x, frontend, backend_fw, test_flags, fn_tree, on_
...
+    dtype_and_x,
+    frontend,
+    test_flags,
+    fn_tree,
+    on_device
+):
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        backend_to_test=backend_fw,
```
Comment (AlexandraPosta) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_fft/test_discrete_fourier_transform.py:277]: Please try to run the tests locally and check the logs, "backend_fw" is not defined

Comment (MartinOcharo) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_fft/test_discrete_fourier_transform.py:277]: Hello @AlexandraPosta! I have added the "backend_fw". However, when I try to run the test I get this ImportError;

"ImportError: cannot import name 'array_helpers_dtype_info_helper' from partially initialized module 'ivy_tests.test_ivy.helpers.hypothesis_helpers.array_helpers' (most likely due to a circular import) (/home/codespace/.local/lib/python3.10/site-packages/ivy-1.1.9-py3.10.egg/ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py)"

Ivy error

**Comment by AlexandraPosta** [comment]
Hello @MartinOcharo, please push your latest changes so that I can have a proper look. When you do so, make sure to resolve the merge conflicts as well. :zap:

In terms of the import error, have you made any changes to other files? It is quite hard to debug because I do not actually know what changes you have made.

Additionally, I had a quick look at your ifft2 and it has a few issues. I have not run the function, but it will fail because you are passing inexistent parameters to ivy.ifft (s) and incorrect data type (dim should be int, but you want to pass on axis which is a tuple). To avoid these issues, please go through [our setup documentation](https://unify.ai/docs/ivy/overview/contributing/setting_up.html) and run the test locally. 

Keep up the work! I hope this helps, happy coding! :cat:

**Pull request closed without merging**